crates/circuit/src/dag_circuit.rs

```diff
@@ -8002,25 +8000,6 @@ impl ::std::ops::Index<NodeIndex> for DAGCircuit {
     }
 }
 
-/// Add to global phase. Global phase can only be Float or ParameterExpression so this
-/// does not handle the full possibility of parameter values.
-/// TODO replace/merge this with add_param/radd_param
```

Comment:
As the comment explains, this was deliberately not added to `Param` because it cannot handle all the variants. This PR doesn't change the situation, it just silently encodes panicking conditions into a method presenting itself as general.

Reply:
Yeah good point -- since we're still relying on adding/multiplying heavily it would still be nice if this would be a method I think. 

We could attempt to implement add/mul by trying to defer to Python and return an error if it didn't work out. But given that we never use this, my preferred way would be to just raise an error if this is used on an `Obj`.

Reply:
If you name them `add_numeric` and `mul_numeric` I'm happy with it.

Reply:
Scalar is fine by me too, yeah - anything that indicates that they aren't fully general.